src/packageRequester.ts

```diff
@@ -269,6 +275,7 @@ async function resolveAndFetch (
         latest,
         normalizedPref,
         resolution,
+        sideEffectsCache: await getSideEffectsCache(ctx.storePath, id, options.sideEffectsCache),
```

Comment:
there is no need to call the function if sideEffectsCache is false, so

```js
sideEffectsCache: options.sideEffectsCache && await getSideEffectsCache(ctx.storePath, id),
```

Reply:
This makes the type signature of `sideEffectsCache` really complex. Why not just calling the method?

Reply:
Thanks for changing it. It wasn't critical but I think code duplication is better than bad abstraction. We will probably refactor it in the future